pkg/serverless/logs/parse.go

```diff
@@ -24,13 +24,19 @@ type platformObjectRecord struct {
 }
 
 // reportLogMetrics contains metrics found in a LogTypePlatformReport log
+// initDurationMs is sent at the very end of the tx
+// but initDurationTelemetry is also the duration in ms
+// and along with initStartTime, are provided by TelemetryAPI
+// early in the invocation, which is a bit confusing
+// TODO Astuyve - refactor out initDurationMs to draw from TelemetryAPI
 type reportLogMetrics struct {
 	durationMs            float64
 	billedDurationMs      int
 	memorySizeMB          int
 	maxMemoryUsedMB       int
 	initDurationMs        float64
 	initDurationTelemetry float64
+	initStartTime         time.Time
```

Comment:
nit: _technically_ I feel like several of these fields (including the one you added) should be moved to their own separate struct. It seems to me that the `startLogItem`, `runtimeDoneItem`, and `reportLogItem` were initially intended to contain information from a single type of log event.

This doesn't need to be fixed. But it did confuse me for a second.

Reply:
You're 10000% right here. We get the initDuration in MS via the report log, which is what this struct was meant to do. I extended it unnaturally when adding support for the telemetry API field here, but I do think I should move it.